packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts

```diff
@@ -95,11 +106,84 @@ export async function forceSourceRetrieveSourcePath(explorerPath: vscode.Uri) {
       return;
     }
   }
+
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
     new FilePathGatherer(explorerPath),
-    new ForceSourceRetrieveSourcePathExecutor(),
+    useBetaRetrieve(explorerPath)
+      ? new LibraryRetrieveSourcePathExecutor()
+      : new ForceSourceRetrieveSourcePathExecutor(),
     new SourcePathChecker()
   );
   await commandlet.run();
 }
+
+// this supported types logic is temporary until we have a way of generating the metadata type from the path
+// once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+export function useBetaRetrieve(explorerPath: vscode.Uri): boolean {
+  const filePath = explorerPath.fsPath;
+  const betaDeployRetrieve = sfdxCoreSettings.getBetaDeployRetrieve();
+  const supportedType =
+    isAuraOrLWC(filePath) ||
+    path.extname(filePath) === APEX_CLASS_EXTENSION ||
+    filePath.includes(`${APEX_CLASS_EXTENSION}-meta.xml`) ||
+    (path.extname(filePath) === APEX_TRIGGER_EXTENSION ||
+      filePath.includes(`${APEX_TRIGGER_EXTENSION}-meta.xml`)) ||
+    (path.extname(filePath) === VISUALFORCE_COMPONENT_EXTENSION ||
+      filePath.includes(`${VISUALFORCE_COMPONENT_EXTENSION}-meta.xml`)) ||
+    (path.extname(filePath) === VISUALFORCE_PAGE_EXTENSION ||
+      filePath.includes(`${VISUALFORCE_PAGE_EXTENSION}-meta.xml`));
+  return betaDeployRetrieve && supportedType;
+}
+
+export function isAuraOrLWC(filePath: string) {
```

Comment:
Is there a better way of doing this ?

Reply:
Maybe something like this?

```typescript
import { RegistryAccess, registryData } from '@salesforce/source-deploy-retrieve'

const registry = new RegistryAccess();
const component = registry.getComponentsFromPath(filePath)[0];
const typeName = component.type.name;
const { auradefinitionbundle, lightningcomponentbundle } = registryData.types;
return typeName === auradefinitionbundle.name ||
       typeName === lightningcomponentbundle.name;
```

Reply:
I like this better since I could replace a lot of the logic in `useBetaRetrieve` that evaluates the paths